### ISSUE TITLE
Unable to see any members in object browser

### ISSUE THREAD

**Issue by MarkBoyer** [halcyon-tech/vscode-ibmi#562]
In the past this has worked fine, however when I expand a saved object browser filter, or create a new one, when I expand the filter no members are shown.  There are no errors in the output (see below).  Did I miss something at some update version of the extension?

/home/wg446mb: system "DSPFD FILE(RM12PTF446/QRPGLESRC) TYPE(*ATR) FILEATR(*PF) OUTPUT(*OUTFILE) OUTFILE(ILEDITOR/O_7m7Ca4qS)"
{
    "code": null,
    "signal": null,
    "stdout": "CPF9861: Output file O_7M7CA4QS created in library ILEDITOR.\nCPF9862: Member O_7M7CA4QS added to output file O_7M7CA4QS in library ILEDITOR.\nCPF3030: 1 records added to member O_7M7CA4QS in file O_7M7CA4QS in ILEDITOR.",
    "stderr": ""
}

/home/wg446mb: echo "SELECT * FROM ILEDITOR.O_7m7Ca4qS" | LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i')"
{
    "code": null,
    "signal": null,
    "stdout": "DB2>",
    "stderr": ""
}

 - Extension version 1.0.1
 - IBM i OS version 7.4

**Comment by MarkBoyer** [comment]
1.  When run by hand it returns the DSPFD information.
2. Turning SQL off returns the member list, however clicking on a member opens an editor page but does not. load the contents from the source member.
3. There is only 1 member.  I did a refresh so I'm assuming the same file in ILEDITOR was used and the contents replaced?
4. Now that members are being displayed, I can see that with SQL turned on, the list of objects, in this case the single source file I have defined in the filter, was not displaying.  So now I can see the object(s), the members when I expand, however I cannot see the contents of a member.  No error, simply:
/home/wg446mb: echo "CREATE OR REPLACE ALIAS ILEDITOR.RM12PTF446_SRCFILE_WSSMTXFL for RM12PTF446.SRCFILE(\"WSSMTXFL\")" | LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i')"
{
    "code": null,
    "signal": null,
    "stdout": "DB2>",
    "stderr": ""
}

/home/wg446mb: echo "select srcdat, rtrim(srcdta) as srcdta from ILEDITOR.RM12PTF446_SRCFILE_WSSMTXFL" | LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i')"
{
    "code": null,
    "signal": null,
    "stdout": "DB2>",
    "stderr": ""
}

/home/wg446mb: echo "CREATE OR REPLACE ALIAS ILEDITOR.RM12PTF446_SRCFILE_WSSMTXFL for RM12PTF446.SRCFILE(\"WSSMTXFL\")" | LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i')"
{
    "code": null,
    "signal": null,
    "stdout": "DB2>",
    "stderr": ""
}

/home/wg446mb: echo "select srcdat, rtrim(srcdta) as srcdta from ILEDITOR.RM12PTF446_SRCFILE_WSSMTXFL" | LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i')"
{
    "code": null,
    "signal": null,
    "stdout": "DB2>",
    "stderr": ""
}

When I run the select statement above by hand, it returns file not found, however when I run the create or replace alias command by hand, it creates fine.

**Comment by worksofliam** [comment]
@MarkBoyer The temp work around is to disable source dates. Thought I know you probably don't want this.

**Comment by worksofliam** [comment]
@MarkBoyer Totally weird! I've tried to connect with you on LinkedIn. Would you be okay to Zoom with me a little so we can troubleshoot this issue?

Thanks!

**Comment by MarkBoyer** [comment]
@worksofliam I accepted on LinkedIn.  I am in the Pacific time zone.  I would be able to have a call between now and 12:00 noon, or between 1:00 and 4:00 pm if either of those work for your schedule.  Thank you for the support!

**Comment by worksofliam** [comment]
Seems like an issue with the SQL program or perhaps some system setup. I am debugging, but here are my notes from the call.

```
-bash-5.1$ system "call QSYS/QZDFMDB2 PARM('SELECT * FROM ILEDITOR.O_2iGoUMwc')"
ODDCEN  ODDDAT  ODDTIM  ODLBNM     ODOBNM     ODOBTP   ODOBAT     ODOBFR  ODOBSZ       ODOBTX                                             ODOBLK  ODOBDM  ODCCEN  ODCDAT  ODCTIM  ODOBOW     ODSCEN  O
------- ------- ------- ---------- ---------- -------- ---------- ------- ------------ -------------------------------------------------- ------- ------- ------- ------- ------- ---------- ------- -
1       022322  095325  RM12PTF446 SRCFILE    *FILE    PF         0           6160384  Wynne Systems Source File                          0       0       1       042220  134624  WG446MB    1       0
  1 RECORD(S) SELECTED.
-bash-5.1$ echo "SELECT * FROM ILEDITOR.O_2iGoUMwc" | system "call QSYS/QZDFMDB2 PARM('-i')"
DB2>
-bash-5.1$
```

* Working: `system "call QSYS/QZDFMDB2 PARM('SELECT * FROM ILEDITOR.O_2iGoUMwc')"`
* Not working: `echo "SELECT * FROM ILEDITOR.O_2iGoUMwc" | system "call QSYS/QZDFMDB2 PARM('-i')"`

**Comment by worksofliam** [comment]
@MarkBoyer Are you able to provide your QCCSID here?

**Comment by worksofliam** [comment]
@MarkBoyer Could you also try this from the pase shell in Code for IBM i?

```
$ echo cat | system call qshell/cat
cat
```

**Comment by MarkBoyer** [comment]
@worksofliam QCCSID is 37 and the command completes without error:
-bash-5.1$ echo cat | system call qshell/cat
-bash-5.1$

**Comment by MarkBoyer** [comment]
@worksofliam It completes:
 $                                 
 echo cat | system call qshell/cat 
 cat                               
 $

**Comment by worksofliam** [comment]
@MarkBoyer So it might be that there is something wrong with the way your descriptors are setup. I don't know a lot about that. BUT, I am planning on pushing out a possible patch in the next hour which I want you to try.

I will post back here if anything else comes up.

**Comment by worksofliam** [comment]
@MarkBoyer I just published 1.0.2 and should be available in 5 or minutes. It might solve your issue.. but it also might not. Give it a whirl and let me know.

**Comment by MarkBoyer** [comment]
@worksofliam Did not resolve:
-bash-5.1$ LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i')"
SELECT * FROM ILEDITOR.O_jAEQsaFs
DB2>
-bash: SELECT: command not found

**Comment by MarkBoyer** [comment]
@worksofliam the above is when I run it manually from an ssh session the VSC output is:
/home/wg446mb: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i')"
SELECT * FROM ILEDITOR.O_jAEQsaFs
{
    "code": null,
    "signal": null,
    "stdout": "DB2>",
    "stderr": ""
}

**Comment by worksofliam** [comment]
@MarkBoyer Super weird. I will keep digging.

**Comment by worksofliam** [comment]
@MarkBoyer Can you try this?

```
/QOpenSys/usr/bin/echo "SELECT * FROM ILEDITOR.O_2iGoUMwc" | system "call QSYS/QZDFMDB2 PARM('-i')"
```

**Comment by MarkBoyer** [comment]
@worksofliam No change, and it looks like that where it knows echo resides:
-bash-5.1$ /QOpenSys/usr/bin/echo "SELECT * FROM ILEDITOR.O_2iGoUMwc" | system "call QSYS/QZDFMDB2 PARM('-i')"
DB2>
-bash-5.1$ which echo
/QOpenSys/pkgs/bin/echo

Just looking around here (https://www.ibm.com/docs/en/i/7.4?topic=utilities-pase-i-system-utility)m does something like this not return the output the way you need it?  This works in terminal.
system -i "CALL QSYS/QZDFMDB2 parm('SELECT * FROM ILEDITOR.O_2iGoUMwc')"

**Comment by worksofliam** [comment]
@MarkBoyer yes, that method does return output. The downside is that it only allows parameters of a certain size. 

I will get back to you ASAP with some more things to try.

**Comment by worksofliam** [comment]
@MarkBoyer Another simple one. Can you try this?

```
echo 'SELECT * FROM ILEDITOR.O_2iGoUMwc' | system "call QSYS/QZDFMDB2 PARM('-i')"
```

**Comment by worksofliam** [comment]
@MarkBoyer Are you also able to provide these files in your home directory if you have them?

* `.profile`
* `.bashrc`
* `.bash_aliases`

**Comment by MarkBoyer** [comment]
@worksofliam - Same result:
-bash-5.1$ echo 'SELECT * FROM ILEDITOR.O_2iGoUMwc' | system "call QSYS/QZDFMDB2 PARM('-i')"
DB2>
-bash-5.1$ 
[Archive.zip](https://github.com/halcyon-tech/vscode-ibmi/files/8129575/Archive.zip)
I've attached my .profile, renames MB.prolfile, and .bash_profile, renames MB.bash_profile.  I don't not have a .bashrc nor .bash_aliases in my home directory.

**Comment by worksofliam** [comment]
@MarkBoyer Next thing to try:

Use `WRKENVVAR` and add `QIBM_USE_DESCRIPTOR_STDIO` at a `*SYS` level to the value of `Y`.

Then restart your Code for IBM i connection and try these two. Let me know which of them work. Make sure the table you're selecting (`ILEDITOR.O_2iGoUMwc`) exists first.

1. echo "SELECT * FROM ILEDITOR.O_2iGoUMwc" | system "call QSYS/QZDFMDB2 PARM('-i')"
2. echo "SELECT * FROM ILEDITOR.O_2iGoUMwc" | system -e "call QSYS/QZDFMDB2 PARM('-i')"

**Comment by worksofliam** [comment]
@MarkBoyer For later: if you are not able to set that environment variable, please give this a try:

```
bash -c "cl 'ADDENVVAR ENVVAR(QIBM_USE_DESCRIPTOR_STDIO) VALUE(Y) REPLACE(*YES)' >/dev/null 2>&1; echo 'SELECT * FROM ILEDITOR.O_2iGoUMwc' | system \"call QSYS/QZDFMDB2 PARM('-i')\""
```

**Comment by MarkBoyer** [comment]
@worksofliam  - no change on results.  Adding the environment variable made no change, just displayed the DB2> prompt then returned to the bash prompt.  I also used system -i ADDENVVAR "ENVVAR(QIBM_USE_DESCRIPTOR_STDIO) VALUE(Y)" to add the environment variable to my bash job, neither of the suggested options returned results.  Also, running the bash command to add the environment variable hangs for a period of time and then returns to the original bash prompt.

**Comment by worksofliam** [comment]
@MarkBoyer Can you confirm that this command didn't work? It just hangs and then returns to the prompt?

```
bash -c "cl 'ADDENVVAR ENVVAR(QIBM_USE_DESCRIPTOR_STDIO) VALUE(Y) REPLACE(*YES)' >/dev/null 2>&1; echo 'SELECT * FROM ILEDITOR.O_2iGoUMwc' | system \"call QSYS/QZDFMDB2 PARM('-i')\""
```

**Comment by worksofliam** [comment]
@MarkBoyer Were able to add the `QIBM_USE_DESCRIPTOR_STDIO=Y` at a `WRKENVVAR *SYS` level from 5250? Also, is that env var already defined in `WRKENVVAR`?

**Comment by MarkBoyer** [comment]
@worksofliam correct, it does not work.  It hangs for about 25 seconds and then returns to the bash prompt.  I just retested and this was without the environment variable, the previous test was with the environment variable.
-bash-5.1$ bash -c "cl 'ADDENVVAR ENVVAR(QIBM_USE_DESCRIPTOR_STDIO) VALUE(Y) REPLACE(*YES)' >/dev/null 2>&1; echo 'SELECT * FROM ILEDITOR.O_2iGoUMwc' | system \"call QSYS/QZDFMDB2 PARM('-i')\""
-bash-5.1$

**Comment by MarkBoyer** [comment]
@worksofliam - yes, I did define the environment variable at the *SYS level, tested and removed.  It was not defined prior to my adding it.

**Comment by worksofliam** [comment]
@MarkBoyer Well, there is likely a much bigger problem here. Do you have access to IBM Support? Might be worth putting in something with them since now this is slightly out of my reach since I don't have access to your system.

If you don't, let me know and I can try get someone else involved who might know the details of this issue.

**Comment by MarkBoyer** [comment]
@worksofliam - Hello sir.  I know it's been ages since we chatted about this, and I just went silent.  We have more developers interested in using VSC and your extension, but everyone gets the same issue when source dates are enabled, which we need.  I can get a call opened with IBM, but I honestly don't know what to tell them.  Do you have any suggestion how we can proceed and try to get this resolved?  TIA

**Comment by worksofliam** [comment]
@MarkBoyer I have sent you a DM on LinkedIn to see if we can setup a meeting. Thanks.

**Comment by worksofliam** [comment]
@yepadoos I think your issue is a bit different from this issue. Notice that the object types are different only those filters (`*SRCPF` vs `*ALL`)

**Comment by yepadoos** [comment]
Thanks, in the past when I had the filter setup for object type *ALL the members were displayed.  I re-added this filter with *SRCPF and it works now.  Thanks

**Comment by zzmand** [comment]
Hi, 
I can see same result in "Code for IBM i" output :
"code": null,
"signal": null,
"stdout": "DB2>",
"stderr": ""

The reason is that SSHD server on LPAR is starting old fashioned way through Autostart job entries. You can check this by
-bash-5.1$ **ps -ef | grep sshd**

if you get something like this :
**SomeUser 120858      1   0 08:38:01      -  0:00 /QOpenSys/usr/sbin/sshd**
you are using Autostart job entries and you can try to start SSHD in regular way **STRTCPSVR SERVER(*SSHD)**

you should have this :
 **qsecofr   121     1   0   Apr 23      -  0:00 /QOpenSys/QIBM/ProdData/SC1/OpenSSH/sbin/sshd**

have a nice day
Dan

**Comment by MarkBoyer** [comment]
@zzmand I finally had a chance to look into your suggestion.  I've got it working now, but almost the opposite of your suggestion, but TBH I don't understand why.

When I first logged in and ran the ps command I got:
sshdusr  858    1   0   May 15      -  0:00 /QOpenSys/usr/sbin/sshd  

So I did ENDTCPSVR and a STRTCPSVT for *SSHD.  I then got:
qsecofr 5427    1   0 11:34:32      -  0:00 /QOpenSys/QIBM/ProdData/SC1/OpenSSH/sbin/sshd

Unfortunately, nothing could connect then and any attempt immediately returned "Connection closed by...".  So, I did ENDTCPSVR again and restarted SSHD with "QSH CMD('/QOpenSys/usr/sbin/sshd')". Then ps returned:
wg446mb 5545    1   0 12:19:45      -  0:00 /QOpenSys/usr/sbin/sshd 

I could connect with SSH and I was able to see al source members, well, once I changed all filters to *SRCPF instead of *ALL.  When I look at the program that is running, it is just a symlink to the one that wouldn't allow login:
lrwxrwxrwx 1 qsys 0 82 Jul 15  2017 /QOpenSys/usr/sbin/sshd -> ../../QIBM/ProdData/SC1/OpenSSH/sbin/sshd  

So, thank you very much for your input, I am up and running again (happy dev dance), but I'm certainly not clear why.  The same daemon is running now as before just under another user.  Sounds like authorities to me, but SSHDUSER has *ALLOBJ, no group or supplemental groups, I did notice it set to *SIGNOFF which I don't think should matter but it also has limited capabilities set to *YES.  Perhaps that's it?  I would have expected the STRTCPSVR under QSECOFR to have worked though.  I would love to understand what the cause of the original failure was so I don't have to keep it running under my profile and maybe it can help someone else in the future.

CC:  @worksofliam
